README.md

```diff
@@ -1,4 +1,45 @@
-## Rusty Merkle Tree Implementation
-**TODO**
-- [ ] Write REAMDE
+# Rusty Merkle Tree Implementation
 
+Simple Merkle tree implementation in Rust.
+
+## Installation
+
+- Clone the repository
+```bash
+git clone https://github.com/LucasUTNFRD/Merkle-Tree.git && cd Merkle-Tree
+```
+
+## Features
+
+- Create Merkle trees from any data that implements `AsRef<[u8]>`
+- Generate Merkle proofs for leaf verification
+- Verify Merkle proofs
+- Add new elements to existing trees
+- Generic implementation supporting different data types
+- Uses SHA3-256 for secure hashing
+
+## API
+
+### Creating a New Merkle Tree
+
+```rust
+let merkle = MerkleTree::new(&data)?;
+```
+
+### Generating a Proof
+
+```rust
+let proof = merkle.generate_proof(&element)?;
+```
+
+### Verifying a Proof
+
+```rust
+let is_valid = merkle.verify_proof(&element, &proof);
+```
+
+### Adding New Elements
+
+```rust
+merkle.add(new_element);
+```
```

Comment:
Lovely example!!! These are always great!

However, what do you think about making a single code block and putting all your code inside it? Then, when people use your library they can simple copy and paste your example.

Plus, you can use the explanation in the subtitles as code comments inside the block. Something like:

```rust
int main() {
    // Create a new Merkle Tree
    let merkle = MerkleTree::new(&data)?;
}
```